Lib/shutil.py

```diff
@@ -958,6 +958,8 @@ def unpack_archive(filename, extract_dir=None, format=None):
     if extract_dir is None:
         extract_dir = os.getcwd()
 
+    filename = os.fspath(filename)
```

Comment:
I notice that the tests show `extract_dir` working for a path-like object, but there's no explicit conversion here. If it's because something else explicitly covers that then please add a comment. Otherwise just explicitly do the `os.fspath()` conversion for `extract_dir`.